src/views/Change.vue

```diff
@@ -202,6 +203,13 @@ export default class Change extends Mixins(CommonMixin, FeeMixin, FilingTemplate
       // fetch entity snapshot
       const entitySnapshot = await this.fetchEntitySnapshot()
 
+      switch (entitySnapshot.businessInfo.legalType) {
+        case CorpTypeCd.BENEFIT_COMPANY:
+        case CorpTypeCd.BC_COMPANY:
+        case CorpTypeCd.BC_CCC:
+        case CorpTypeCd.BC_ULC_COMPANY:
+          throw new Error(`Invalid entity type, must not be benefit company id`)
+      }
```

Comment:
This looks like a good place to check this 👍 

However, this could fail if we add new entity types and forget to check this code. (Why would we? This is for a firm change filing.)

1. I think it would be better to throw the error if the current legal type is NOT a firm (SP or GP).
2. Can `entitySnapshot` be null? Can `entitySnapshow.businessInfo` be null? If so then you can use optional chaining (*) to prevent a de-reference error.

(*) We should use optional chaining sparingly, since it can mask errors that we might want to otherwise see or handle.

Reply:
i have implemented both of these changes in the new commit

Reply:
PS Let reviewers resolve the conversations they started -- it's a "bookmark" for them to review their comments. Otherwise they have to re-review the entire code and try to remember what they commented on and see if the code is fixed.